rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -62,6 +69,47 @@ SequentialCompressionWriter::~SequentialCompressionWriter()
 
 void SequentialCompressionWriter::compression_thread_fn()
 {
+  if (compression_options_.thread_nice_value) {
+
+#ifdef _WIN32
+    ROSBAG2_COMPRESSION_LOG_WARN_STREAM(
+      "Lowering of process priority is not implemented for windows");
+
+    /**
+     * The implementation should look something like this
+
+    uint8_t nice_value = *compression_options_.thread_nice_value;
+
+    // this must match THREAD_PRIORITY_IDLE, THREAD_PRIORITY_LOWEST...
+    DWORD dwThreadPri = *compression_options_.thread_nice_value;
+
+    if(!SetThreadPriority(GetCurrentThread(), dwThreadPri))
+    {
+      ROSBAG2_COMPRESSION_LOG_WARN_STREAM(
+        "Could not set nice value of compression thread to " << nice_value << " : " << std::strerror(GetLastError()));
+    }
+    */
+
+#else
+    int wanted_nice_value = *compression_options_.thread_nice_value;
+
+    errno = 0;
+    int cur_nice_value = getpriority(PRIO_PROCESS, 0);
+    if (cur_nice_value != -1 && errno == 0) {
```

Comment:
@jmachowinski I need to double check yet. But it seems here mistake in the if condition.
The point is that `cur_nice_value == -1` is a valid return value from the `getpriority(..)`
It seems the correct logic shall be `if (!(new_nice_value == -1 && errno != 0))` or its equivalent `if (new_nice_value != -1 || errno != 0)`

Reply:
good catch, this is indeed wrong

Reply:
fixed